desktop/src/main.rs

```diff
@@ -128,17 +129,39 @@ fn load_movie_from_path(
 }
 
 fn run_player(opt: Opt) -> Result<(), Box<dyn std::error::Error>> {
-    let movie_url = if opt.input_path.exists() {
-        let absolute_path = opt
-            .input_path
-            .canonicalize()
-            .unwrap_or_else(|_| opt.input_path.to_owned());
-        Url::from_file_path(absolute_path)
-            .map_err(|_| "Path must be absolute and cannot be a URL")?
-    } else {
-        Url::parse(opt.input_path.to_str().unwrap_or_default())
-            .map_err(|_| "Input path is not a file and could not be parsed as a URL.")?
+    let movie_url = match &opt.input_path {
+        Some(path) => {
+            if path.exists() {
+                let absolute_path = path.canonicalize().unwrap_or_else(|_| path.to_owned());
+                Url::from_file_path(absolute_path)
+                    .map_err(|_| "Path must be absolute and cannot be a URL")?
+            } else {
+                Url::parse(path.to_str().unwrap_or_default())
+                    .map_err(|_| "Input path is not a file and could not be parsed as a URL.")?
+            }
+        }
+        None => {
+            let result = open_file_dialog(Option::from("swf"), None).unwrap_or_else(|e| {
+                panic!(e);
```

Comment:
Let's not panic here and instead return the error upward with `?`. (`map_err` if you need to turn it into a different type of error).